### PR TITLE
Support setting depth to structured logging

### PR DESCRIPTION
**Pull request by yangjunmyfm192085**
Signed-off-by: JunYang <yang.jun22@zte.com.cn>

**Which issue(s) this PR fixes**:

Fixes #https://github.com/kubernetes/kubernetes/issues/102353

### CODE REVIEW

File: contributors/devel/sig-instrumentation/migration-to-structured-logging.md [hunk 1]
```diff
@@ -95,50 +103,21 @@ Structured logging functions follow a different logging interface design than ot
...
 
 Simplified mapping between functions:
-* `klog.Infof`, `klog.Info`, `klog.Infoln`, `klog.InfoDepth` -> `klog.InfoS`
+* `klog.Infof`, `klog.Info`, `klog.Infoln` -> `klog.InfoS`
+* `klog.InfoDepth` -> `klog.InfoSDepth`
 * `klog.V(N).Infof`, `klog.V(N).Info`, `klog.V(N).Infoln` -> `klog.V(N).InfoS`
-* `klog.Warning`, `klog.Warningf`, `klog.Warningln`, `klog.WarningDepth` -> `klog.InfoS`
+* `klog.Warning`, `klog.Warningf`, `klog.Warningln` -> `klog.InfoS`
+* `klog.WarningDepth` -> `klog.InfoSDepth`
 * `klog.V(N).Warning`, `klog.V(N).Warningf`, `klog.V(N).Warningln`, `klog.V(N).WarningDepth` -> `klog.V(N).InfoS`
```
Comment (serathius) [inline contributors/devel/sig-instrumentation/migration-to-structured-logging.md:111]: Let's use this occasion to clean up the incorrect examples from the original guide to make it more readable. There is no `klog.V(N).*` other then for `Info*`. There are not `klog.V(N).Warning*`, no `klog.V(N).Error*` and no `klog.V(N).Fatal`. Can you remove them ?

Comment (yangjunmyfm192085) [reply contributors/devel/sig-instrumentation/migration-to-structured-logging.md:111]: OK, Let me do it

**Comment by serathius** [comment]
Thanks, otherwise everything looks ok. 
/cc @timhockin @DirectXMan12 @pohly

**Comment by yangjunmyfm192085** [comment]
I have fixed it @serathius, thanks.

**Comment by serathius** [comment]
/approve
Let's test if permissions are taken from master

**Comment by serathius** [comment]
Nope, make sense. 
@yangjunmyfm192085, I can approve this under condition that you cherry-pick your change on new master branch to pick up https://github.com/kubernetes/community/pull/5804

**Pull request merged**